src/App.js

```diff
@@ -1,85 +1,141 @@
 import React from 'react';
+import NewTodo from './NewTodo';
+import TodoList from './TodoList';
+import TodosFilters from './TodosFilter';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-4" />
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todos: [],
+    selectedFilter: 'All',
+  };
+
+  addNewTodo = (newTodo) => {
+    if (newTodo.trim() === '') {
+      return;
+    }
+
+    this.setState(state => ({
+      todos: [
+        ...state.todos,
+        {
+          id: +new Date(),
+          title: newTodo,
+          completed: false,
+        },
+      ],
+    }));
+  };
+
+  changeTodoStatus = (todoId, status) => {
+    this.setState(state => ({
```

Comment:
```suggestion
    this.setState(prevState => ({
```
В этом объекте хранится предыдущее состояние state